File: backend/app/constants/preferences.py
from enum import Enum


class InvestorType(str, Enum):
    """Supported investor types."""

    HODLER = "hodler"
    DAY_TRADER = "day_trader"
    NFT_COLLECTOR = "nft_collector"


class ContentType(str, Enum):
    """Supported content types."""

    MARKET_NEWS = "market_news"
    CHARTS = "charts"
    SOCIAL = "social"
    FUN = "fun"



COINGECKO_ASSET_IDS = {
    "BTC": "bitcoin",
    "ETH": "ethereum",
    "SOL": "solana",
    "XRP": "ripple",
    "ADA": "cardano",
}

SUPPORTED_ASSETS = list(COINGECKO_ASSET_IDS.keys())

File: frontend/src/types/preferences.ts
export type AssetSymbol = 'BTC' | 'ETH' | 'SOL' | 'XRP' | 'ADA'

export type InvestorType = 'hodler' | 'day_trader' | 'nft_collector'

export type ContentType = 'market_news' | 'charts' | 'social' | 'fun'

export type PreferenceCreate = {
  assets: AssetSymbol[]
  investor_type: InvestorType
  content_types: ContentType[]
}

export type PreferenceUpdate = {
  assets?: AssetSymbol[]
  investor_type?: InvestorType
  content_types?: ContentType[]
}

export type Preference = {
  user_id: number
  assets: AssetSymbol[]
  investor_type: InvestorType
  content_types: ContentType[]
}

export const SUPPORTED_ASSETS: AssetSymbol[] = [
  'BTC',
  'ETH',
  'SOL',
  'XRP',
  'ADA',
]

export const SUPPORTED_INVESTOR_TYPES: InvestorType[] = [
  'hodler',
  'day_trader',
  'nft_collector',
]

export const SUPPORTED_CONTENT_TYPES: ContentType[] = [
  'market_news',
  'charts',
  'social',
  'fun',
]
